### DISCUSSION TITLE
Unsupported origin issue MV3

### DISCUSSION THREAD

**Discussion by thebinod7** [InboxSDK/InboxSDK#926] [General]
Hi, I am using this library in my project built using react along with webpack but Im not being able to initialize the library. Please check the screenshots and let me know what am I missing.  I get this error '**unsupported origin'**

![content](https://user-images.githubusercontent.com/9801330/209794110-f2fc4939-6a7b-4a57-ba73-7dbdc73d46d4.png)
![error](https://user-images.githubusercontent.com/9801330/209794116-ed99cf38-6860-47e9-b94c-d2b3dcee7be1.png)
![mv3](https://user-images.githubusercontent.com/9801330/209794117-82c9b22f-8846-48f4-9343-fe750c47c027.png)

Thanks!

**Comment by bbest31** [comment]
This library also does not work for me either when using an extension built via react and webpack. The content script is being read fine, but the callback on InboxSDK.load() is never executed for me.

**Comment by thebinod7** [comment]
Need to configure webpack config:
`entry: {
		popup: './src/popup.js',
		background: './src/background.js',
		pageWorld: '@inboxsdk/core/pageWorld.js',
		inboxsdk: './src/content/inboxsdk.js',
	},`

And add this code to your background.js file
`chrome.runtime.onMessage.addListener((message, sender, sendResponse) => {
	if (message.type === 'inboxsdk__injectPageWorld' && sender.tab) {
		if (chrome.scripting) {
			// MV3
			chrome.scripting.executeScript({
				target: { tabId: sender.tab.id },
				world: 'MAIN',
				files: ['pageWorld.js']
			});
			sendResponse(true);
		} else {
			// MV2 fallback. Tell content script it needs to figure things out.
			sendResponse(false);
		}
	}
});`
